### PR TITLE
Added autoFocus feature for the input field in ModusTextInput component

### LINKED ISSUE #649: Modus Text Input - Add autoFocus for the input field and a method to set the focus when needed
**Describe the solution you'd like**
Add a prop `auto-focus-input` and a public method `focusInput` to `modus-text-input` component.

So that I can focus the input field when needed.

### PR DESCRIPTION
**Pull request by msankaran0712**
## Description

Added a prop `auto-focus-input` and a public method `focusInput` to `modus-text-input` component.

The current application is the `modus-tree-view-item` component will use `auto-focus-input` to set the initial focus on the text field when actions like add item and edit item are performed.

Fixes #649 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

Tested on Chrome browser.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (including CHANGELOG updates)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by ryan-henness-trimble** [comment]
Two small additional changes:

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ View all releases at: https://github.com/trimble-oss/modus-web-components/releas
 - Modus Chips active state color is now blue
 - Remove unused Warning button and make Primary button the default
 
+### Added
```
Comment (ryan-henness-trimble) [inline CHANGELOG.md:14]: See comments in discussion tab

Comment (msankaran0712) [reply CHANGELOG.md:14]: @ryan-henness-trimble is there any change needed in this file?

File: stencil-workspace/storybook/stories/components/modus-text-input/modus-text-input.stories.tsx [hunk 2]
```diff
@@ -9,44 +9,51 @@ export default {
...
-        type: { summary: 'string' }
-      }
+        type: { summary: 'string' },
+      },
+    },
+    autoFocusInput: {
+      name: 'auto-focus-input',
+      description: 'Sets autofocus for the input',
+      table: {
+        type: { summary: 'string' },
```
Comment (ryan-henness-trimble) [inline stencil-workspace/storybook/stories/components/modus-text-input/modus-text-input.stories.tsx:19]: should this be a boolean?

File: CHANGELOG.md (hunk 1 above)
Comment (ryan-henness-trimble) [reply CHANGELOG.md:15]: nope nope, sorry. It wouldn't let me submit review without a comment on an actual file

**Comment by msankaran0712** [comment]
@ryan-henness-trimble I guess all the comments are fixed now, please check now

**Review by ryan-henness-trimble — APPROVED** [review]

**Pull request merged**